### PR TITLE
screwdriver.Emitter to emit logs to file

### PR DESCRIPTION
**Pull request by nkatzman**
screwdriver.Emitter is an io.WriteCloser that will be used to read stdout/error and write it to a file

### CODE REVIEW

File: launch_test.go [hunk 1]
```diff
@@ -833,3 +882,34 @@ func TestMergePRError(t *testing.T) {
...
+		return nil
+	}
+
+	// oldMkdirAll := mkdirAll
+	// defer func() { mkdirAll = oldMkdirAll }()
+	// mkdirAll = os.MkdirAll
+
+	called := false
+
+	newEmitter = func(path string) (screwdriver.Emitter, error) {
```
Comment (Filbird) [inline launch_test.go]: Should there be a defer to set this to its original value?

File: launch_test.go [hunk 2]
```diff
@@ -833,3 +882,34 @@ func TestMergePRError(t *testing.T) {
...
+		return &MockEmitter{
+			close: func() error {
+				called = true
+				return nil
+			},
+		}, nil
+	}
+
+	if err := launchAction(screwdriver.API(api), "testBuildID", testRoot); err != nil {
```
Comment (Filbird) [inline launch_test.go]: Why are we calling launchAction rather than launch?

**Comment by Filbird** [comment]
👍  🚀

**Comment by d2lam** [comment]
Is it possible to add some comments to the function header so it's easier to review?

**Comment by jer** [comment]
@d2lam which function are you wanting extra comments on? the exported ones have documentation

**Comment by d2lam** [comment]
@jer  it's okay. Talked to Noah in person so he explained the code to me 😄

**Comment by stjohnjohnson** [comment]
LGTM 👍  
jack_is_desperate_for_friends__and_even_tries_talking_to_hot_dogs

**Comment by d2lam** [comment]
LGTM 👍

**Pull request merged**